packages/frontend-plugin-api/src/components/ExtensionBoundary.tsx

```diff
@@ -56,6 +61,13 @@ export interface ExtensionBoundaryProps {
 /** @public */
 export function ExtensionBoundary(props: ExtensionBoundaryProps) {
   const { node, routable, children } = props;
+  const componentsApi = useApi(componentsApiRef);
+
+  const plugin = node.spec.source;
+  const Progress = componentsApi.getComponent(coreComponentsRefs.progress);
```

Comment:
I think we can add a `useComponentRef` already that makes this a single step instead. Can be a followup though

Reply:
Create an issue in our board, will do in a follow-up PR